pkgs/applications/editors/vscode/extensions/default.nix

```diff
@@ -4209,16 +4209,20 @@ let
         mktplcRef = {
           name = "shellcheck";
           publisher = "timonwong";
-          version = "0.26.3";
-          sha256 = "GlyOLc2VrRnA50MkaG83qa0yLUyJYwueqEO+ZeAStYs=";
+          version = "0.37.0";
+          sha256 = "1d0blynn6c2hz4y9fk7b5wsa3x168gxyycr5d05zqp0rx520m5wc";
         };
         nativeBuildInputs = [ jq moreutils ];
         postInstall = ''
           cd "$out/$installPrefix"
           jq '.contributes.configuration.properties."shellcheck.executablePath".default = "${shellcheck}/bin/shellcheck"' package.json | sponge package.json
         '';
         meta = {
```

Comment:
```suggestion
        meta = with lib; {
```

Reply:
Using `with` is bad practice. It creates problems in static analysis. This has been debated many times. The recommendation is to not use `with`. Particularly on `meta`. But for `meta.maintainers` is fine.

Reply:
https://nix.dev/guides/best-practices.html

Reply:
Not sure about whether this is "lots of code". While I do agree about the nested `with` making the namespace unclear, can you link me any discussion where consent has been reached or an official statement on whether we want to avoid `with lib; meta`? I see that in your [tracking issue](https://github.com/NixOS/nixpkgs/issues/208242), you state the nix.dev article, which (afais) only talks about top-level `with`s, which this is not

Reply:
AFAIK the consensus is:
* nested `with` is discouraged. (And is being slowly being removed form nixpkgs.)
* simultaneously using `with` in `meta` and `meta.maintainers` is nested with.
* using `with` in maintainers is acceptable as long as meta doesn't have `with`.
* not using `with` in maintainers is acceptable.

I have no further information to add atm.

Reply:
https://github.com/NixOS/nixpkgs/issues/292468#issuecomment-2016454242

Reply:
Thank you! This is indeed a valid point, I wasn't aware of this discussion. I will use this in my next PRs then. :)